### PR TITLE
Settings controller [MAILPOET-1757]

### PR DESCRIPTION
**Pull request by costasovo**
I'm very sorry for such a looooong PR. Most of it are just replacements of static calls.
It should be reviewable when going commit by commit

After this gets merged we have to use this controller instead of static Setting::getValue/setValue.
This has to be rebased and checked/upadted for some new occurrences of Setting::xxx calls before merging.

### CODE REVIEW

File: lib/Settings/SettingsController.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    $settings = self::$settings;
+    $setting =& $settings;
+    foreach($keys as $key) {
+      $setting =& $setting[$key];
+      if(!is_array($setting)) {
+        $setting = [];
+      }
+    }
+    $setting[$last_key] = $value;
+    Setting::setValue($main_key, $settings[$main_key]);
```
Comment (pavel-mailpoet) [inline lib/Settings/SettingsController.php]: Should we move `Setting::setValue` and `Setting::getValue` to here to discourage people from using it?

File: lib/Config/Populator.php [hunk 2]
```diff
@@ -154,13 +157,13 @@ private function createDefaultSettings() {
...
-    if(!Setting::getValue('sender')) {
-      Setting::setValue('sender', $sender);
+    if(!$this->settings->fetch('sender')) {
+      $this->settings->set('sender', $sender);
     }
 
     // enable signup confirmation by default
-    if(!Setting::getValue('signup_confirmation')) {
-      Setting::setValue('signup_confirmation', array(
+    if(!$this->settings->fetch('signup_confirmation')) {
```
Comment (pavel-mailpoet) [inline lib/Config/Populator.php]: I don't understand why do we need to use `fetch` in this class.

File: lib/Settings/SettingsController.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    $this->ensureLoaded();
+    return self::$settings;
+  }
+
+  function set($key, $value) {
+    $this->ensureLoaded();
+    $keys = explode('.', $key);
+    $main_key = $keys[0];
+    $last_key = array_pop($keys);
+    $settings = self::$settings;
```
Comment (pavel-mailpoet) [inline lib/Settings/SettingsController.php]: I don't think we need this variable here.

File: lib/Newsletter/ViewInBrowser.php [hunk 4]
```diff
@@ -2,12 +2,19 @@
...
 use MailPoet\Newsletter\Links\Links;
 use MailPoet\Newsletter\Renderer\Renderer;
 use MailPoet\Newsletter\Shortcodes\Shortcodes;
 
 class ViewInBrowser {
+
+  /** @var bool */
+  private $is_tracking_enabled;
+
+  function __construct($is_tracking_enabled) {
```
Comment (pavel-mailpoet) [inline lib/Newsletter/ViewInBrowser.php]: How is this working? I don't see passing the argument anywhere.

Comment (costasovo) [reply lib/Newsletter/ViewInBrowser.php]: It is passed here: https://github.com/mailpoet/mailpoet/pull/1766/commits/e7dfdf1cdbfd354ed1084a53f4d1f13f6b6b711e#diff-a565a70a1dc7bccd9aec5961b86160f9R35

File: lib/Settings/SettingsController.php (hunk 1 above)
Comment (costasovo) [reply lib/Settings/SettingsController.php]: Very good point!

**Comment by costasovo** [comment]
I've moved handling defaults into new controller and also moved static setValue, getValue as a private methods to the controller so that they can be used any more...
also rebased on current master

**Comment by costasovo** [comment]
Rebased on current master

**Pull request merged**